processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -79,13 +86,39 @@ public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    throw new UnsupportedOperationException();
+    return factory.makeValueMatcher(dimension, new Predicate<String>()
+    {
+      @Override public boolean apply(String input)
+      {
+        // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+        return value.equals(fn.apply(Strings.emptyToNull(input)));
+      }
+    });
   }
 
   @Override
-  public ValueMatcher makeMatcher(ColumnSelectorFactory factory)
+  public ValueMatcher makeMatcher(ColumnSelectorFactory columnSelectorFactory)
   {
-    throw new UnsupportedOperationException();
+    final DimensionSelector dimensionSelector = columnSelectorFactory.makeDimensionSelector(dimension, null);
+    if (dimensionSelector == null) {
+      return new BooleanValueMatcher(Strings.isNullOrEmpty(fn.apply(value)));
```

Comment:
if a column was missing, should we ever match it in this case? fn.apply(value) looks a little bit unusual as value is really an element from the Range of the function and not Domain. may be it should be...

```
return new BooleanValueMatcher(false);
```

@xvrl ?